### PR TITLE
feat: add FE charts

### PR DESCRIPTION
**Pull request by gi0baro**
### Changes summary

- add CI linting step
- add `fe-static` chart
- add `fe-nextjs` chart

### CODE REVIEW

File: charts/fe-nextjs/templates/ingress.yaml [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+  {{- $_ := set .Values.ingress.annotations "kubernetes.io/ingress.class" .Values.ingress.className}}
+  {{- end }}
+{{- end }}
+{{- if semverCompare ">=1.19-0" .Capabilities.KubeVersion.GitVersion -}}
+apiVersion: networking.k8s.io/v1
+{{- else if semverCompare ">=1.14-0" .Capabilities.KubeVersion.GitVersion -}}
+apiVersion: networking.k8s.io/v1beta1
+{{- else -}}
+apiVersion: extensions/v1beta1
+{{- end }}
```
Comment (edubacco) [inline charts/fe-nextjs/templates/ingress.yaml:15]: The oldest k8s version we use it's 1.18, I think `else v1beta1` is enough

File: charts/fe-nextjs/templates/proxy.yaml [hunk 2]
```diff
@@ -0,0 +1,91 @@
+{{- if .Values.staticProxy.enabled }}
```
Comment (edubacco) [inline charts/fe-nextjs/templates/proxy.yaml:1]: I do not get the point here, why a next.js application should proxy something? how do we use it?

File: charts/fe-nextjs/templates/proxy.yaml [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+{{- if semverCompare ">=1.19-0" .Capabilities.KubeVersion.GitVersion -}}
+apiVersion: networking.k8s.io/v1
+{{- else if semverCompare ">=1.14-0" .Capabilities.KubeVersion.GitVersion -}}
+apiVersion: networking.k8s.io/v1beta1
+{{- else -}}
+apiVersion: extensions/v1beta1
+{{- end }}
+kind: Ingress
+metadata:
+  name: {{ $fullName }}-static-proxy
```
Comment (edubacco) [inline charts/fe-nextjs/templates/proxy.yaml:48]: as it's the only ingress of the chart, why adding to the name "static-proxy"?

File: charts/fe-nextjs/templates/_helpers.tpl [hunk 4]
```diff
@@ -0,0 +1,77 @@
+{{/*
```
Comment (edubacco) [inline charts/fe-nextjs/templates/_helpers.tpl:1]: I imagined in this file also a section defining common labels, as per the standard [casavo labelling rules](https://casavo.atlassian.net/wiki/spaces/cd/pages/1435467786/Kubernetes+Labeling)

**Review by edubacco — COMMENTED** [review]
I watched just `fe-nextjs`, I leave comments only for that part :)

File: charts/fe-nextjs/templates/_helpers.tpl [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+app.kubernetes.io/managed-by: {{ .Release.Service }}
+{{- with .Values.labels }}
+{{ toYaml . }}
+{{- end }}
+{{- end }}
+
+{{/*
+Selector labels
+*/}}
+{{- define "app.proxySelectorLabels" -}}
```
[the remark this answers is not in the document]
Comment (gi0baro) [reply charts/fe-nextjs/templates/_helpers.tpl:74]: yup.

File: charts/fe-nextjs/templates/ingress.yaml (hunk 1 above)
Comment (gi0baro) [reply charts/fe-nextjs/templates/ingress.yaml:15]: Ok, I'll also change compatibility version in `Chart.yaml` then.

File: charts/fe-nextjs/templates/proxy.yaml (hunk 2 above)
Comment (gi0baro) [reply charts/fe-nextjs/templates/proxy.yaml:1]: See https://github.com/casavo/lisa/blob/master/helm/values/production.yaml#L33-L49

File: charts/fe-nextjs/templates/proxy.yaml (hunk 3 above)
Comment (gi0baro) [reply charts/fe-nextjs/templates/proxy.yaml:48]: it is not.. https://github.com/casavo/charts/pull/1/files#diff-d24f275219f1bc1a6bca2abe899c83748039de12d81893c2692889d0e638ce9a

File: charts/fe-nextjs/templates/_helpers.tpl (hunk 4 above)
Comment (gi0baro) [reply charts/fe-nextjs/templates/_helpers.tpl:1]: How should I know those labels in advance? Will be added in values by projects IMHO

Comment (gi0baro) [reply charts/fe-nextjs/templates/_helpers.tpl:1]: @edubacco also: remember all of this is **public** ;)

Comment (gi0baro) [reply charts/fe-nextjs/templates/_helpers.tpl:1]: Let's talk about abstraction layer for required labels in CoP Devops

**Review by edubacco — APPROVED** [review]

**Pull request merged**